numpy/core/overrides.py

```diff
@@ -45,12 +45,21 @@ def get_overloaded_types_and_args(relevant_args):
             # By default, insert this argument at the end, but if it is
             # subclass of another argument, insert it before that argument.
             # This ensures "subclasses before superclasses".
-            index = len(overloaded_args)
-            for i, old_arg in enumerate(overloaded_args):
-                if issubclass(arg_type, type(old_arg)):
-                    index = i
-                    break
-            overloaded_args.insert(index, arg)
+            # (Note: avoid setting up the iterator for the first item,
+            # since that usually will be the only one.)
+            if overloaded_args:
+                index = len(overloaded_args)
+                for i, old_arg in enumerate(overloaded_args):
+                    if issubclass(arg_type, type(old_arg)):
+                        index = i
+                        break
+                overloaded_args.insert(index, arg)
+            else:
+                overloaded_args = [arg]
+
+    # Short-cut for the common case of only ndarray.
+    if overloaded_types == [ndarray]:
```

Comment:
Should we save this list instead of recreating it?

Reply:
Done.